src/uct/ib/base/ib_iface.c

```diff
@@ -370,6 +390,53 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+static int uct_ib_iface_is_same_roce_info(uct_ib_roce_version_desc_t *local_roce_ver,
+                                          const uct_ib_address_t *remote_ib_addr)
+{
+    int ret, remote_roce_ver;
+
+    if ((local_roce_ver->addr_family == AF_INET6) &&
+        (remote_ib_addr->flags & UCT_IB_FLAG_ROCE_IPV6)) {
+        goto check_roce_ver;
+    } else {
+        if ((local_roce_ver->addr_family == AF_INET) &&
+            !(remote_ib_addr->flags & UCT_IB_FLAG_ROCE_IPV6)) {
+            goto check_roce_ver;
+        } else {
+            ucs_assert(local_roce_ver->addr_family != 0);
+            ucs_debug("different addr_family detected. local %s remote %s",
+                      (local_roce_ver->addr_family == AF_INET6 ?
+                       "AF_INET6" : "AF_INET"),
+                      (remote_ib_addr->flags & UCT_IB_FLAG_ROCE_IPV6 ?
+                       "AF_INET6" : "AF_INET4"));
+            return 0;
+        }
+    }
+
+check_roce_ver:
```

Comment:
can be separate function

Reply:
it's a pretty short check and used only here and the entire function isn't long...

Reply:
but it does exact operation and can be called from 2 places instead of `goto`

Reply:
ok